core/templates/pages/blog-dashboard-page/blog-post-editor/blog-post-editor.component.spec.ts

```diff
@@ -0,0 +1,537 @@
+    expect(component.MAX_CHARS_IN_BLOG_POST_TITLE).toBe(
+      AppConstants.MAX_CHARS_IN_BLOG_POST_TITLE);
+    expect(component.initEditor).toHaveBeenCalled;
+    expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+    expect(component.DEFAULT_PROFILE_PICTURE_URL).toEqual(defaultImageUrl);
+    expect(windowDimensionsService.isWindowNarrow()).toHaveBeenCalled();
+    expect(component.windowIsNarrow).toBe(true);
+  });
+
+  it('should get schema', () => {
+    expect(component.getSchema())
+      .toEqual(component.HTML_SCHEMA);
+  });
+
+  it('should return true for header enabled callback', () => {
+    expect(component.headersAreEnabledCallBack()).toEqual(true);
+  });
+
+  it('should successfully fetch blog post editor data', fakeAsync(() => {
+    let blogPostEditorData = {
+      username: 'test_user',
+      profilePictureDataUrl: 'sample_url',
+      listOfDefaulTags: ['news', 'Learners'],
+      maxNumOfTags: 2,
+      blogPostDict: sampleBlogPostData,
+    };
+    component.blogPostId = 'sampleBlogId';
+    spyOn(blogPostEditorBackendApiService, 'fetchBlogPostEditorData')
+      .and.returnValue(Promise.resolve(blogPostEditorData));
+
+    component.initEditor();
+    tick();
+
+    expect(blogPostEditorBackendApiService.fetchBlogPostEditorData)
+      .toHaveBeenCalled();
+    expect(component.authorUsername).toEqual('test_user');
+    expect(component.blogPostData).toEqual(sampleBlogPostData);
+    expect(component.authorProfilePictureUrl).toEqual('sample_url');
+    expect(component.defaultTagsList).toEqual(['news', 'Learners']);
+    expect(component.maxAllowedTags).toEqual(2);
+    expect(component.thumbnailDataUrl).toEqual(
+      '/assetsdevhandler/blog_post/sampleBlogId/assets/thumbnail' +
+      '/image.png');
+    expect(component.dateTimeLastSaved).toEqual(
+      'November 21, 2014 at 04:52 AM');
+    expect(component.title).toEqual('sample_title');
+    expect(component.contentEditorIsActive).toEqual(false);
+    expect(component.lastChangesWerePublished).toBe(true);
+  }));
+
+  it('should display alert when unable to fetch blog post editor data',
+    fakeAsync(() => {
+      spyOn(blogPostEditorBackendApiService, 'fetchBlogPostEditorData')
+        .and.returnValue(Promise.reject({status: 500}));
+      spyOn(alertsService, 'addWarning');
+
+      component.initEditor();
+      tick();
+
+      expect(blogPostEditorBackendApiService.fetchBlogPostEditorData)
+        .toHaveBeenCalled();
+      expect(alertsService.addWarning).toHaveBeenCalledWith(
+        'Failed to get blog post data.');
+    }));
+
+  it('should update local title value', () => {
+    spyOn(blogPostUpdateService, 'setBlogPostTitle');
+    component.title = 'Sample title changed';
+
+    component.blogPostData = sampleBlogPostData;
+    component.updateLocalTitleValue();
+
+    expect(blogPostUpdateService.setBlogPostTitle).toHaveBeenCalledWith(
+      component.blogPostData, component.title);
```

Comment:
It might be better to put the value rather than the variable to make sure the test is more robust.

Reply:
Done.